pkg/scheduler/service/bookkeepingtask.go

```diff
@@ -0,0 +1,78 @@
+package service
+
+import (
+	"fmt"
+	"github.com/kyma-incubator/reconciler/pkg/model"
+	"github.com/pkg/errors"
+	"go.uber.org/zap"
+	"time"
+)
+
+type BookkeepingTask interface {
+	Apply(reconResult *ReconciliationResult, maxRetries int) []error
+}
+
+type markOrphanOperation struct {
+	transition *ClusterStatusTransition
+	logger     *zap.SugaredLogger
+}
+
+func (oo markOrphanOperation) Apply(reconResult *ReconciliationResult, maxRetries int) []error {
```

Comment:
If you don't need maxRetries here, then we should not design this `Apply` as a shared interface method.

Reply:
Decided to pass bk config to bk tasks